File: climbing_stairs.java
class Solution {
    public int climbStairs(int n) {
        if(n < 0)
        return 0;
    if(n == 1)
        return 1;

    int[] store = new int[n];

    store[0] = 1;
    store[1] = 2;

    for(int i = 2; i < n; ++i)
        store[i] = store[i-1] + store[i-2];

    return store[n-1];
    }
}

File: climbing_stairs.rb
def climb_stairs(n)
    res = [0, 1]
    n.times do
        res << (res[-1] + res[-2])
    end
    res.last
end

def climb_stairs(n)
    return 0 if n == 0
  a, b = 0, 1
  n.times { a, b = b, b + a }
  b
end
